data-service/src/main/java/org/ihtsdo/buildcloud/service/PermissionService.java

```diff
@@ -0,0 +1,84 @@
+package org.ihtsdo.buildcloud.service;
+
+import org.ihtsdo.buildcloud.entity.ReleaseCenter;
+import org.ihtsdo.sso.integration.SecurityUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
+
+@Service
+public class PermissionService {
+
+    private final Logger logger = LoggerFactory.getLogger(PermissionService.class);
+
+    public final static String GLOBAL_SUFFIX = "_GLOBAL";
+
+    public static final String GLOBAL_ROLE_SCOPE = "global";
+
+    public enum Role {
+        ADMIN, RELEASE_MANAGER, USER, AUTHOR
+    }
+
+    @Autowired
+    private PermissionServiceCache permissionServiceCache;
+
+    @Autowired
+    private ReleaseCenterService releaseCenterService;
+
+    public boolean userHasRoleOnReleaseCenter(String role, String releaseCenterKey) {
+        boolean contains = false;
+        if (releaseCenterKey.equalsIgnoreCase(GLOBAL_ROLE_SCOPE)) {
+            Set<String> globalRoles = permissionServiceCache.getGlobalRoles(SecurityUtil.getAuthenticationToken());
+            contains = globalRoles.contains(role);
+        } else {
+            ReleaseCenter releaseCenter = releaseCenterService.find(releaseCenterKey);
+            Map<String, Set<String>> rolesToCodeSystemMap = permissionServiceCache.getCodeSystemRoles(SecurityUtil.getAuthenticationToken());
+            if (!StringUtils.isEmpty(releaseCenter.getCodeSystem()) && rolesToCodeSystemMap.containsKey(releaseCenter.getCodeSystem())) {
+                Set<String> roles = rolesToCodeSystemMap.get(releaseCenter.getCodeSystem());
+                if (Role.USER.name().equalsIgnoreCase(role)) {
+                    contains = roles.contains(Role.USER.name()) || roles.contains(Role.AUTHOR.name());
+                } else {
+                    contains = roles.contains(role);
+                }
+            }
+        }
+
+        if (!contains) {
+            logger.info("User '{}' does not have required role '{}' on release center '{}'.", SecurityUtil.getUsername(), role, releaseCenterKey);
+        }
+        return contains;
+    }
+
+    public Map getRolesForLoggedInUser() {
```

Comment:
This method seems more complicated than it needs to be and it also will not allow new roles to be added without code changes in this class.
Could I suggest that you simply create a map with a key of the code system and a value of all the roles the user has on that code system. There is no need to hardcode specific roles here, just take them all as strings?
You could add another key in the map for "global" in addition to the code systems?
I think this would be clearer and simpler to maintain.

Reply:
The logic has been updated to simplify the way of getting roles. Please have a look @kaicode

Reply:
This is design is much easier to comprehend 👍